tests/framework/extensions/workloads/create.go

```diff
@@ -0,0 +1,62 @@
+package workloads
+
+import (
+	steveV1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	containerName     = "nginx"
+	containerImage    = "nginx"
+	defaultNamespace  = "default"
+	isCattleLabeled   = true
+	port              = "port"
+	serviceAppendName = "service-"
+	ServiceType       = "service"
+)
+
+// CreateDeployment is a helper function to create a deployment and service in the downstream cluster.
+func CreateDeployment(steveclient *steveV1.Client, wlName string) (*steveV1.SteveAPIObject, *steveV1.SteveAPIObject, error) {
+	containerTemplate := NewContainer(containerName, containerImage, corev1.PullAlways, []corev1.VolumeMount{}, []corev1.EnvFromSource{}, nil, nil, nil)
+	podTemplate := NewPodTemplate([]corev1.Container{containerTemplate}, []corev1.Volume{}, []corev1.LocalObjectReference{}, nil)
+	deployment := NewDeploymentTemplate(wlName, defaultNamespace, podTemplate, isCattleLabeled, nil)
```

Comment:
Same comment as ingress, have this function take the templates rather than happening here. We want to make this extension flexible to use.

Reply:
Spoke offline, addressed this.